Two SSL-related pynntp pytests fail

https://github.com/cclauss/apt-get-inn2-docker/actions/runs/11018800364/job/30599897391#step:8:121
```
=========================== short test summary info ============================
FAILED tests/integration/inn2/test_inn2.py::test_nntp_client_with_ssl - ssl.SSLError: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1000)
FAILED tests/integration/inn2/test_inn2.py::test_nntp_client_with_ssl_starttls - nntp.nntp.NNTPPermanentError: 580 Error initializing TLS
=================== 2 failed, 38 passed, 14 xfailed in 1.47s ===================
```
@greenbender, do you have any insight into how to fix these two failing tests?
* `Generating SSL key` startup step is missing.

This repo tries to be as close to yours but is built on `debian:bookworm` and uses `apt-get install inn2`.

---
% `docker build --no-cache --progress plain -t inn . && docker run --rm -it -p119:119 -p563:563 inn`

> #10 12.58 Setting up inn2-inews (2.7.1-1+deb12u1) ...
#10 12.58
#10 12.58 Configuration file '/etc/news/inn.conf'
#10 12.58  ==> File on system created by you or by a script.
#10 12.58  ==> File also in package provided by package maintainer.
#10 12.58  ==> Using current old file as you requested.
[ ... ]
#10 13.65 Setting up openssl (3.0.14-1~deb12u2) ...
[ ... ]
#10 13.80 Setting up ca-certificates (20230311) ...
[ ... ]
#10 14.74 Setting up inn2 (2.7.1-1+deb12u1) ...
#10 14.80
#10 14.80 Configuration file '/etc/news/readers.conf'
#10 14.80  ==> File on system created by you or by a script.
#10 14.80  ==> File also in package provided by package maintainer.
#10 14.80  ==> Using current old file as you requested.
#10 14.81 Installing initial content for /var/lib/news/active
#10 14.81 Installing initial content for /var/lib/news/newsgroups
#10 14.81 Building history database in /var/lib/news... makehistory: tradspool: mapping file /var/spool/news/tradspool.map not found

--- a/Dockerfile
+++ b/Dockerfile
@@ -12,8 +12,6 @@ RUN apt-get update -qq && \
   sed -i 's/#\(tlscapath:\)/\1/' etc/news/inn.conf && \
   sed -i 's/#\(tlscertfile:\)/\1/' etc/news/inn.conf && \
   sed -i 's/#\(tlskeyfile:\)/\1/' etc/news/inn.conf && \
-  touch /etc/key.pem && \
-  chown news:news /etc/key.pem && \
   mkdir -p /var/run/news && \
   touch /var/run/news/nnrpd-563.pid && \
   chown news:news /var/run/news/nnrpd-563.pid && \

--- a/docker-entrypoint.sh
+++ b/docker-entrypoint.sh
@@ -1,11 +1,11 @@
 #!/bin/sh
 
-if [ ! -f etc/key.pem ]; then
-	echo "Generating SSL key"
-	openssl req -x509 -newkey rsa:4096 -sha256 -days 3650 -nodes -keyout etc/key.pem -out etc/cert.pem -subj "/CN=news.localhost"
-fi
-
 . lib/news/innshellvars
+
+if [ ! -f $TLSKEYFILE ]; then
+	echo "Generating SSL key"
+	openssl req -x509 -newkey rsa:4096 -sha256 -days 3650 -nodes -keyout $TLSKEYFILE -out $TLSCERTFILE -subj "/CN=news.localhost"
+fi
 
 # Support for implicit TLS
 nnrpd -D -p 563 -S

